src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,41 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("user should be not null");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("user has null login");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("user has null password");
+        }
+        if (user.getAge() == null) {
+            throw new InvalidDataException("user has null age");
+        }
+        if (user.getLogin().length() < 1) {
+            throw new InvalidDataException("user have to have login");
```

Comment:
Let's improve the message 
```suggestion
            throw new InvalidDataException("user should have login longer then " + LOGIN_SIZE);
```